htdp-lib/test-engine/racket-tests.rkt

```diff
@@ -109,9 +109,9 @@
 
 (define (do-check-expect test expected src)
   (error-check (lambda (v) (if (number? v) (exact? v) #t))
-               expected INEXACT-NUMBERS-FMT #t)
+               expected INEXACT-NUMBERS-FMT #t (list expected expected))
```

Comment:
I don't understand why `expected` is listed twice. This feels wrong.

Reply:
I changed the error message to:

    check-expect: cannot compare inexact numbers, but the second argument is ~a. Try (check-within test ~a error-range).

Where the "`, but the second argument is ~a`" part is new.
So there are two `~a` referring to `expected` now.